apps/files/src/components/FilesLists/StatusIndicators/StatusIndicators.vue

```diff
@@ -0,0 +1,75 @@
+<template>
+  <div class="uk-flex uk-flex-middle">
+    <DefaultIndicators
+      v-if="displayDefaultIndicators"
+      :item="item"
+      :currentFolderPath="currentFolderPath"
+      :class="{ 'uk-margin-xsmall-right' : customIndicators }"
+    />
+    <component
+      v-if="customIndicators"
+      :is="customIndicators"
+      :item="item"
+      :currentFolderPath="currentFolderPath"
+    />
+  </div>
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+
+const DefaultIndicators = () => import('./DefaultIndicators.vue')
+
+export default {
+  name: 'StatusIndicators',
+
+  components: {
+    DefaultIndicators
+  },
+
+  props: {
+    item: {
+      type: Object,
+      required: true
+    },
+    currentFolderPath: {
+      type: String,
+      required: true
+    }
+  },
+
+  data () {
+    return {
+      customIndicators: false
+    }
+  },
+
+  computed: {
+    ...mapGetters(['configuration']),
+
+    displayDefaultIndicators () {
+      return !this.configuration.theme.filesList.hideDefaultStatusIndicators
+    }
+  },
+
+  created () {
+    this.loadCustomIndicators()
+  },
+
+  methods: {
+    loadCustomIndicators () {
+      const path = `/themes/${this.configuration.currentTheme}/status-indicators.js`
+
+      requirejs([path], this.injectCustomIndicators, this.onRequireError)
```

Comment:
Thinking now about the 404 that this will cause in case the file does not exist (which will be most of the time IMHO). I don't really want to do extra call which would first check for the file and then load it in case it exists. So what about another key in theme json which would define if the load should happen or not? If it would be true we can then expect that the file is there or it then makes sense to throw a 404. But it's a small extra effort for administrators in the end. Seems to me like a way to go though. What do you think @PVince81 @kulmann @DeepDiver1975 ?

Reply:
In the beginning, I was going to approach it as "just another extension" but then it wasn't really feeling right separating the theme-related stuff into two different places. Talking about simple or complex I think it should live at some united place not to make it complicated to look around where it actually lives. And looking at the complex theming requirement, most of it will live in the json theme file only as a value anyway.

Otherwise, I would simply call it an extension and wouldn't mix it into the theme. But then I'd say this could be done overall and we could drop simple/complex theming and just have theming and then extensions.

Reply:
Result of discussion with @PVince81 - we won't consider this a theming but it will be an extension/customisation. We'll reuse logic for loading extensions. (ref https://github.com/owncloud/phoenix/issues/2404#issuecomment-579673241)